### PR TITLE
Changes to raise exceptions for unsupported storage files

### PR DESCRIPTION
**Pull request by Onager**

### CODE REVIEW

File: plaso/cli/psteal_tool.py [hunk 1]
```diff
@@ -186,9 +186,9 @@ def AnalyzeEvents(self):
     storage_reader = storage_factory.StorageFactory.CreateStorageReaderForFile(
         self._storage_file_path)
     if not storage_reader:
-      logger.error('Format of storage file: {0:s} not supported'.format(
-          self._storage_file_path))
-      return
+      raise errors.BadConfigOption(
+          'Format of storage file: {0:s} not supported'.format(
+              self._storage_file_path))
```
Comment (joachimmetz) [inline plaso/cli/psteal_tool.py]: add the "Format of storage file: {0:s} not supported for writing" check?

File: plaso/cli/psteal_tool.py [hunk 2]
```diff
@@ -186,9 +186,9 @@ def AnalyzeEvents(self):
     storage_reader = storage_factory.StorageFactory.CreateStorageReaderForFile(
         self._storage_file_path)
     if not storage_reader:
-      logger.error('Format of storage file: {0:s} not supported'.format(
-          self._storage_file_path))
-      return
+      raise errors.BadConfigOption(
```
Comment (joachimmetz) [inline plaso/cli/psteal_tool.py]: please update docstring as well

File: plaso/cli/psort_tool.py [hunk 3]
```diff
@@ -512,9 +512,9 @@ def ProcessStorage(self):
     storage_reader = storage_factory.StorageFactory.CreateStorageReaderForFile(
         self._storage_file_path)
     if not storage_reader:
-      logger.error('Format of storage file: {0:s} not supported'.format(
-          self._storage_file_path))
-      return
+      raise errors.BadConfigOption(
```
Comment (joachimmetz) [inline plaso/cli/psort_tool.py]: please update docstring as well

File: plaso/cli/psteal_tool.py (hunk 1 above)
Comment (Onager) [reply plaso/cli/psteal_tool.py]: I'm not sure what you mean here - this is only opened for reading.

File: plaso/cli/psteal_tool.py (hunk 2 above)
Comment (Onager) [reply plaso/cli/psteal_tool.py]: This already has "BadConfigOption: when a configuration parameter fails validation." and the format of the file provided as an option is being checked - what do you want the in the docstring here?

File: plaso/cli/psort_tool.py (hunk 3 above)
Comment (Onager) [reply plaso/cli/psort_tool.py]: This already has "BadConfigOption: when a configuration parameter fails validation." and the format of the file provided as an option is being checked - what do you want the in the docstring here?

File: plaso/cli/psteal_tool.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/cli/psteal_tool.py]: ack I thought it was the similar path as psort

File: plaso/cli/psort_tool.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/cli/psort_tool.py]: Maybe:
BadConfigOption: when a configuration parameter fails validation or the storage file cannot be opened with write access.

File: plaso/cli/psteal_tool.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/cli/psteal_tool.py]: Maybe:
BadConfigOption: when a configuration parameter fails validation or the storage file cannot be opened with read access.

**Comment by joachimmetz** [comment]
extract and tag end2end test seems to be failing on FC

File: plaso/cli/psort_tool.py (hunk 3 above)
Comment (Onager) [reply plaso/cli/psort_tool.py]: Done

File: plaso/cli/psteal_tool.py (hunk 2 above)
Comment (Onager) [reply plaso/cli/psteal_tool.py]: Done

**Comment by Onager** [comment]
Ping for @joachimmetz

**Pull request merged**